### DISCUSSION TITLE
请问doh地址必须填写IP形式的么..

### DISCUSSION THREAD

**Discussion by yeezylife** [IrineSistiana/mosdns#28] [Q&A] [answered]
我想用dnspod的,于是这样填写了,上不去网..

`  - tag: forward_local # 转发至本地服务器的插件
    type: forward
    args:
      upstream:
        - addr: https://doh.pub/dns-query
    bootstrap:
        - addr: https://9.9.9.11/dns-query`

**Comment by IrineSistiana** [comment, marked as answer]
~bootstrap 的缩进错了。对齐错地方了。没载入。~

**Reply by IrineSistiana** [reply]
额不是，看错了。缩进没错。

连不上你需要看 log。

**Reply by yeezylife** [reply]
> 额不是，看错了。缩进没错。
> 
> 连不上你需要看 log。

我发的好像也有点问题 我的bootstrap:是和args:对齐的,下面是日志

**Reply by IrineSistiana** [reply]
>  Get "https://9.9.9.11:443/dns-query?dns=shEBAAABAAAAAAAAB2FuZHJvaWQHY2xpZW50cwZnb29nbGUDY29tAAABAAE\": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

https://9.9.9.11/dns-query 无响应，像是被墙了。

**Reply by yeezylife** [reply]
> > Get "[https://9.9.9.11:443/dns-query?dns=shEBAAABAAAAAAAAB2FuZHJvaWQHY2xpZW50cwZnb29nbGUDY29tAAABAAE\](https://9.9.9.11:443/dns-query?dns=shEBAAABAAAAAAAAB2FuZHJvaWQHY2xpZW50cwZnb29nbGUDY29tAAABAAE%5C)": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
> 
> https://9.9.9.11/dns-query 无响应，像是被墙了。

我一直用的这个,然后我的remote服务器也是用的这个,我用一模一样的设置,重启之后现在能上网了.
之前还以为是我填的upstream格式有问题

**Reply by IrineSistiana** [reply]
能提供日志么，因为 forward 是用的 adhomeguard 的模块，不是 mosdns 的。所以要看日志说了什么。

**Reply by IrineSistiana** [reply]
看起来像是 adhomeguard bootstrap 的问题。你可以用`ip_addr`手动设 IP。不用 bootstrap。反正 dns.pod 的 IP 从来没变过。我也这么干的。还节省一次不必要的请求。

**Reply by yeezylife** [reply]
我填写的服务器都是我这里可用,其他软件也能用的. 是我bootstrap写法错了?  我把bootstrap: 跟args:对齐,mosdns才打得开..     把bootstrap: 跟upstream:对齐,install service会提示运行失败
哦 我试试写ip_addr,  dns.pod用162.14.21.56和162.14.21.178, 看能不能在系统DNS服务器为127.0.0.1 和 ::1的情况下上网

**Reply by IrineSistiana** [reply]
>  我把bootstrap: 跟args:对齐,mosdns才打得开。跟upstream:对齐,install service会提示运行失败

不对啊，应该 bootstrap 跟 upstream 对齐才对。你的配置文件可能已经乱套了。

不要用记事本编辑配置。用 notepad++ 之类的检查一下缩进。

**Reply by yeezylife** [reply]
..我是复制的你的推荐设置(第二个),然后只改了服务器和bootstrap 和upstream对齐的时候mosdns根本开不起来...

**Reply by yeezylife** [reply]
我去下notepad++ ...

**Reply by IrineSistiana** [reply]
```yaml
  - tag: forward_local # 转发至本地服务器的插件
    type: forward
    args:
      upstream:
        - addr: https://doh.pub/dns-query
      bootstrap:
        - https://223.5.5.5/dns-query
```
刚刚我试了。缩进和 bootstrap 都没问题。

**Reply by yeezylife** [reply]
我重启了一下,能正常用了..感谢感谢感谢..

**Reply by yeezylife** [reply]
我从头到尾在notepad++编写的都不行,开不起来..必须复制你这个才可以..

**Comment by yeezylife** [comment]
"Note: Wiki 里的配置示例之前有很长一段时间里面的匹配器使用了 ! 取反，但是忘记了加 " " 引号。这不会报错，但会导致这些匹配器实际上没有生效。如果之前直接复制并使用了配置示例请留意。"
其实我发现之前的配置示例,如果把forward改成fast_forward,又会出奇奇怪怪的问题,比如dot能用,doh不能用,不过我不敢讲,我怕又是自己把缩进搞错了.  用了你更新的示例之后,用一模一样的设置,fast_forward的doh能正常用了.
